clang/lib/Sema/SemaInit.cpp

```diff
@@ -6585,6 +6585,18 @@ void InitializationSequence::InitializeFrom(Sema &S,
       }
     }
 
+    if (S.getLangOpts().HLSL && Initializer && isa<ConstantArrayType>(DestAT)) {
```

Comment:
Would be nice to consolidate this if statement with the one below, unless HLSL Arrays can always be copy initialized, then it might be nicer left separate.

Reply:
Looking at this a bit. `canPerformArrayCopy` fails in cases that would otherwise be valid in HLSL. At least for HLSL today we can always copy an array if the source and destination type are the same, so we don't need to check that.